custom_components/sat/mqtt/ems.py

```diff
@@ -0,0 +1,146 @@
+
+    @property
+    def supports_setpoint_management(self) -> bool:
+        return True
+
+    @property
+    def supports_hot_water_setpoint_management(self) -> bool:
+        return True
+
+    @property
+    def supports_maximum_setpoint_management(self) -> bool:
+        return True
+
+    @property
+    def supports_relative_modulation_management(self) -> bool:
+        return True
+
+    @property
+    def device_active(self) -> bool:
+        return self.data.get(DATA_CENTRAL_HEATING) == DATA_ON
+
+    @property
+    def flame_active(self) -> bool:
+        return self.data.get(DATA_FLAME_ACTIVE) == DATA_ON
+
+    @property
+    def hot_water_active(self) -> bool:
+        return self.data.get(DATA_DHW_ENABLE) == DATA_ON
+
+    @property
+    def setpoint(self) -> float | None:
+        return float_value(self.data.get(DATA_CONTROL_SETPOINT))
+
+    @property
+    def hot_water_setpoint(self) -> float | None:
+        return float_value(self.data.get(DATA_DHW_SETPOINT))
+
+    @property
+    def boiler_temperature(self) -> float | None:
+        return float_value(self.data.get(DATA_BOILER_TEMPERATURE))
+
+    @property
+    def return_temperature(self) -> float | None:
+        return float_value(self.data.get(DATA_RETURN_TEMPERATURE))
+
+    @property
+    def relative_modulation_value(self) -> float | None:
+        return float_value(self.data.get(DATA_REL_MOD_LEVEL))
+
+    @property
+    def boiler_capacity(self) -> float | None:
+        return float_value(self.data.get(DATA_BOILER_CAPACITY))
+
+    @property
+    def minimum_relative_modulation_value(self) -> float | None:
+        return float_value(self.data.get(DATA_REL_MIN_MOD_LEVEL))
+
+    @property
+    def maximum_relative_modulation_value(self) -> float | None:
+        return float_value(self.data.get(DATA_MAX_REL_MOD_LEVEL_SETTING))
+
+    @property
+    def member_id(self) -> int | None:
+        # Not supported (yet)
+        return None
+
+    async def boot(self) -> SatMqttCoordinator:
+        # Nothing needs to be booted (yet)
+        return self
+
+    def get_tracked_entities(self) -> list[str]:
+        return [DATA_BOILER_DATA]
+
+    async def async_set_control_setpoint(self, value: float) -> None:
+        await self._publish_command(f'{{"cmd": "selflowtemp", "value": {0 if value == 10 else value}}}')
+
+        await super().async_set_control_setpoint(value)
+
+    async def async_set_control_hot_water_setpoint(self, value: float) -> None:
+        await self._publish_command(f'{{"cmd": "dhw/seltemp", "value": {value}}}')
+
+        await super().async_set_control_hot_water_setpoint(value)
+
+    async def async_set_control_thermostat_setpoint(self, value: float) -> None:
+        # Not supported (yet)
+        await super().async_set_control_thermostat_setpoint(value)
+
+    async def async_set_heater_state(self, state: DeviceState) -> None:
+        if (state == DeviceState.ON) != self.device_active:
```

Comment:
Looks like there is a problem with this solution. It checks heatingactive property, but it is also in 'off' state, when the forceheating is set to off, but the current flow temp overrun the currently configured setpoint, so the burner is turned off on the boiler.
In other words, forceheatingoff is still set to off, so requesting for heat, but temporarily no actual vurning happens, only the pump runs. And as a result, the earlier problematic interference with this setting still occurs.
This check should explicitly go against the currently forceheatingoff data in my opinion.

Reply:
```suggestion
if self.desired_heating_active is not None and (state == DeviceState.ON) != self.desired_heating_active:
```